### PR TITLE
PS-6098,PS-6099,PS-6100: Fix skipped parallel replication MTR tests (5.6)

### PR DESCRIPTION
**Pull request by inikep**
PS-6098: MTR tests are skipped because of missing "--slave-parallel-workers="
https://jira.percona.com/browse/PS-6098

Some MTR tests are skipped because of missing `--slave-parallel-workers=#`
To fix the issue:
1. Add `--slave-parallel-workers=2` to `*-slave.opt` files
2. `slave_parallel_workers` should be checked on a slave therefore move
`--source include/only_mts_slave_parallel_workers.inc` or
after
`--source include/rpl_connection_slave.inc`

***

PS-6100: Parallel MTR tests use "--slave-parallel-workers=0"
https://jira.percona.com/browse/PS-6100
    
Make sure that tests with "parallel" and "mts" in the name use parallel replication.

### CODE REVIEW

**Comment by inikep** [comment]
Tested at https://ps56.cd.percona.com/job/percona-server-5.6-param/104/

File: mysql-test/suite/rpl/t/rpl_row_mts_show_relaylog_events-slave.opt [hunk 1]
```diff
@@ -0,0 +1 @@
+--slave-parallel-workers=2 --slave-transaction-retries=0
```
Comment (percona-ysorokin) [inline mysql-test/suite/rpl/t/rpl_row_mts_show_relaylog_events-slave.opt:1]: Is `--slave-transaction-retries=0` required here?

File: mysql-test/suite/rpl/t/rpl_stm_mix_mts_show_relaylog_events-slave.opt [hunk 2]
```diff
@@ -0,0 +1 @@
+--slave-parallel-workers=2 --slave-transaction-retries=0
```
Comment (percona-ysorokin) [inline mysql-test/suite/rpl/t/rpl_stm_mix_mts_show_relaylog_events-slave.opt:1]: Is `--slave-transaction-retries=0` required here?

File: mysql-test/suite/rpl/t/rpl_mts_execute_partial_trx_with_auto_pos_off-master.opt [hunk 3]
```diff
@@ -0,0 +1 @@
+--gtid-mode=ON --enforce-gtid-consistency=ON --log-bin --log-slave-updates
```
Comment (percona-ysorokin) [inline mysql-test/suite/rpl/t/rpl_mts_execute_partial_trx_with_auto_pos_off-master.opt:1]: Is switching GTID mode intentional here?
If yes, then may be moving to `rpl_gtid` suite?

File: mysql-test/suite/rpl/t/rpl_mts_execute_partial_trx_with_auto_pos_off-slave.opt [hunk 4]
```diff
@@ -1 +1 @@
---slave-transaction-retries=0
+--slave-parallel-workers=2 --gtid-mode=ON --enforce-gtid-consistency=ON --slave-transaction-retries=0
```
Comment (percona-ysorokin) [inline mysql-test/suite/rpl/t/rpl_mts_execute_partial_trx_with_auto_pos_off-slave.opt:1]: Likewise. Is switching GTID mode intentional here?

File: mysql-test/suite/rpl/t/rpl_mts_execute_partial_trx_with_auto_pos_on-master.opt [hunk 5]
```diff
@@ -0,0 +1 @@
+--gtid-mode=ON --enforce-gtid-consistency=ON --log-bin --log-slave-updates
```
Comment (percona-ysorokin) [inline mysql-test/suite/rpl/t/rpl_mts_execute_partial_trx_with_auto_pos_on-master.opt:1]: Likewise

File: mysql-test/suite/rpl/t/rpl_mts_execute_partial_trx_with_auto_pos_on-slave.opt [hunk 6]
```diff
@@ -1 +1 @@
---slave-transaction-retries=0
+--slave-parallel-workers=2 --gtid-mode=ON --enforce-gtid-consistency=ON --slave-transaction-retries=0
```
Comment (percona-ysorokin) [inline mysql-test/suite/rpl/t/rpl_mts_execute_partial_trx_with_auto_pos_on-slave.opt:1]: Likewise

File: mysql-test/suite/rpl/t/rpl_row_mts_show_relaylog_events-slave.opt (hunk 1 above)
Comment (inikep) [reply mysql-test/suite/rpl/t/rpl_row_mts_show_relaylog_events-slave.opt:1]: Without it there is a warning:
```
Warnings:
Note	1753	slave_transaction_retries is not supported in multi-threaded slave mode. In the event of a transient failure, the slave will not retry the transaction and will stop.
```

File: mysql-test/suite/rpl/t/rpl_stm_mix_mts_show_relaylog_events-slave.opt (hunk 2 above)
Comment (inikep) [reply mysql-test/suite/rpl/t/rpl_stm_mix_mts_show_relaylog_events-slave.opt:1]: Likewise

File: mysql-test/suite/rpl/t/rpl_mts_execute_partial_trx_with_auto_pos_off-master.opt (hunk 3 above)
Comment (inikep) [reply mysql-test/suite/rpl/t/rpl_mts_execute_partial_trx_with_auto_pos_off-master.opt:1]: `Test requires GTID_MODE=ON`

Comment (inikep) [reply mysql-test/suite/rpl/t/rpl_mts_execute_partial_trx_with_auto_pos_off-master.opt:1]: There is no `rpl_gtid` in 5.6 and 5.7

Comment (percona-ysorokin) [reply mysql-test/suite/rpl/t/rpl_mts_execute_partial_trx_with_auto_pos_off-master.opt:1]: Oh, my bad. I missed that this is 5.6

**Review by percona-ysorokin — APPROVED** [review]
LGTM

**Pull request merged**